src/raiden_libs/matrix.py

```diff
@@ -274,3 +272,51 @@ def _handle_message(self, room: Any, event: dict) -> bool:
             self.message_received_callback(message)
 
         return True
+
+    def join_global_rooms(
+        self, client: GMatrixClient, available_servers: Sequence[str] = ()
+    ) -> None:
+        """Join or create a global public room with given name on all available servers.
+        If can't, create a public room with the name on each server.
+
+        Params:
+            client: matrix-python-sdk client instance
+            servers: optional: sequence of known/available servers to try to find the room in
+        """
+        room_alias_prefix = make_room_alias(self.chain_id, self.service_room_suffix)
+
+        parsed_servers = [
+            urlparse(s).netloc for s in available_servers if urlparse(s).netloc not in {None, ""}
+        ]
+
+        for server in parsed_servers:
+            room_alias_full = f"#{room_alias_prefix}:{server}"
+            log.debug("Trying to join pathfinding room", room_alias_full=room_alias_full)
+            try:
+                broadcast_room = client.join_room(room_alias_full)
+                log.debug("Joined pathfinding room", room=broadcast_room)
+                self.broadcast_rooms.append(broadcast_room)
+            except MatrixRequestError as ex:
+                if ex.code != 404:
+                    log.debug(
+                        "Could not join pathfinding room, trying to create one",
+                        room_alias_full=room_alias_full,
+                    )
+                    try:
+                        global_room = client.create_room(room_alias_full, is_public=True)
+                        log.debug("Created pathfinding room", room=broadcast_room)
+                        self.broadcast_rooms.append(global_room)
+                    except MatrixRequestError:
+                        log.debug(
+                            "Could neither join nor create a global room",
+                            room_alias_full=room_alias_full,
+                        )
+                        TransportError("Could neither join nor create a global room")
```

Comment:
Why create this and then raise it below, can't we do this in one line?

Reply:
+1 ... also since the log string and the exception string is the same perhaps add it in a variable and stuff it into both?